### PR TITLE
Added Configuration Test Helper for Integration Tests using SFA.DAS.AutoConfiguration

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using SFA.DAS.AutoConfiguration;
+
+namespace SFA.DAS.Testing.Helpers
+{
+    public class ConfigurationTestHelper
+    {
+        public static T GetConfiguration<T>(string rowKey)
+        {
+            var configurationService = CreateConfigurationService();
```
Comment (Winsto) [inline SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: Is there a reason for creating  new configuration service on each request?

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            return configurationService.Get<T>(rowKey);
```
Comment (Winsto) [inline SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: could the created configuration service ever be null?

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs (hunk 1 above)
Comment (Winsto) [inline SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: Purely a personal style thing, but (depending on the answers to my other questions) I probably wouldn't bother with explicit local `var`, would probably go for : -
```
CreateConfigurationService()
.Get<T>(rowKey)
```

Just my preference though :)

Comment (tom-gough) [reply SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: no, I will fix

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs (hunk 2 above)
Comment (tom-gough) [reply SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: no

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs (hunk 1 above)
Comment (tom-gough) [reply SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: done

Comment (tom-gough) [reply SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: have made changes according to your other questions, have a look

File: SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace SFA.DAS.Testing.Helpers
+{
+    public class ConfigurationTestHelper
+    {
+        private static IAutoConfigurationService _configurationService;
+
+        public static T GetConfiguration<T>(string rowKey)
+        {
+            if(_configurationService == null)
```
Comment (Winsto) [inline SFA.DAS.Testing/SFA.DAS.Testing/Helpers/ConfigurationTestHelper.cs]: This is starting to look a bit Singleton (ugh! :) ) like, if so is it worth doing it as a proper thread safe singleton? 

I'd suggest the sixth version of [Skeet's implementation](https://csharpindepth.com/Articles/Singleton)

**Pull request closed without merging**